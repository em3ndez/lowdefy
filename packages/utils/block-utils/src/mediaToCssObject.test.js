/*
  Copyright 2020-2024 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import mediaToCssObject from './mediaToCssObject.js';

test('no object', () => {
  expect(mediaToCssObject()).toEqual([]);
});

test('object with null obj to pass', () => {
  const obj = null;
  expect(mediaToCssObject(obj)).toEqual([]);
});

test('object with no media unchanged', () => {
  const obj = {
    a: 'a',
    b: 1,
    c: { a: 'b' },
  };
  expect(mediaToCssObject(obj)).toEqual([obj]);
});

test('array of media objects passed to emotion', () => {
  const obj = [
    {
      a: 'a',
      b: 1,
      c: { a: 'b' },
    },
    {
      a: 'x',
      b: 4,
    },
  ];
  expect(mediaToCssObject(obj)).toEqual(obj);
});

test('string style definitions', () => {
  const obj = 'background-color: red;';
  expect(mediaToCssObject(obj)).toEqual(obj);
});

test('array of string style definitions', () => {
  const obj = ['background-color: red;', 'font-size: 12px;'];
  expect(mediaToCssObject(obj)).toEqual(obj);
});

test('array of mixed style definitions', () => {
  const obj = ['background-color: red;', { value: 'one' }];
  expect(mediaToCssObject(obj)).toEqual(obj);
});

test('object with all media', () => {
  const obj = {
    a: 'a',
    xs: { a: 'xs' },
    sm: { a: 'sm' },
    md: { a: 'md' },
    lg: { a: 'lg' },
    xl: { a: 'xl' },
    xxl: { a: 'xxl' },
  };
  expect(mediaToCssObject(obj)).toMatchSnapshot();
});

test('object with all media with styleObjectOnly option', () => {
  const obj = {
    a: 'a',
    xs: { a: 'xs' },
    sm: { a: 'sm' },
    md: { a: 'md' },
    lg: { a: 'lg' },
    xl: { a: 'xl' },
    xxl: { a: 'xxl' },
  };
  expect(mediaToCssObject(obj, true)).toMatchSnapshot();
});

test('string with all media', () => {
  const obj = `{
    a: 'a',
    @media xs { a: 'xs' }
    @media sm  { a: 'sm' }
    @media md{ a: 'md' }
    @media lg { a: 'lg' }
    @media xl{ a: 'xl' }
    @media xxl  { a: 'xxl' }
  }`;
  expect(mediaToCssObject(obj)).toMatchSnapshot();
});

test('array of mixed types with all media', () => {
  const obj = [`{a: 'a', @media xs { a: 'xs' }}`, 1, true, { xl: { a: 'xl' } }];
  expect(mediaToCssObject(obj)).toEqual([
    `{a: 'a', @media screen and (max-width: 576px) { a: 'xs' }}`,
    {},
    {},
    {
      '@media screen and (min-width: 1200px)': {
        a: 'xl',
      },
    },
  ]);
});
