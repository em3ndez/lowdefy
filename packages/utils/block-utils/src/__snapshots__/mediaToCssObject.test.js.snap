// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object with all media 1`] = `
Array [
  Object {
    "@media screen and (max-width: 576px)": Object {
      "a": "xs",
    },
    "@media screen and (min-width: 1200px)": Object {
      "a": "xl",
    },
    "@media screen and (min-width: 1600px)": Object {
      "a": "xxl",
    },
    "@media screen and (min-width: 576px)": Object {
      "a": "sm",
    },
    "@media screen and (min-width: 768px)": Object {
      "a": "md",
    },
    "@media screen and (min-width: 992px)": Object {
      "a": "lg",
    },
    "a": "a",
  },
]
`;

exports[`object with all media with styleObjectOnly option 1`] = `
Array [
  Object {
    "@media screen and (maxWidth: 576px)": Object {
      "a": "xs",
    },
    "@media screen and (minWidth: 1200px)": Object {
      "a": "xl",
    },
    "@media screen and (minWidth: 1600px)": Object {
      "a": "xxl",
    },
    "@media screen and (minWidth: 576px)": Object {
      "a": "sm",
    },
    "@media screen and (minWidth: 768px)": Object {
      "a": "md",
    },
    "@media screen and (minWidth: 992px)": Object {
      "a": "lg",
    },
    "a": "a",
  },
]
`;

exports[`string with all media 1`] = `
"{
    a: 'a',
    @media screen and (max-width: 576px) { a: 'xs' }
    @media screen and (min-width: 576px) { a: 'sm' }
    @media screen and (min-width: 768px) { a: 'md' }
    @media screen and (min-width: 992px) { a: 'lg' }
    @media screen and (min-width: 1200px) { a: 'xl' }
    @media screen and (min-width: 1600px) { a: 'xxl' }
  }"
`;
